src/mesh/abaqus_io.C

```diff
@@ -852,30 +852,49 @@ void AbaqusIO::read_sideset(std::string sideset_name, sideset_container_t & cont
   dof_id_type elem_id=0;
   unsigned side_id=0;
   char c;
-  std::string dummy;
+  std::string elset_name, dummy;
 
   // Read until the start of another section is detected, or EOF is encountered
   while (_in.peek() != '*' && _in.peek() != EOF)
     {
-      // The strings are of the form: "391, S2"
+      // The strings are either of the form: "391, S2" or "Elset_1, S3"
 
       // Read the element ID and the leading comma
       _in >> elem_id >> c;
+      if (_in)
+      {
+        // For the form of "391, S2"
+        // Read another character (the 'S') and finally the side ID
+        _in >> c >> side_id;
 
-      // Read another character (the 'S') and finally the side ID
-      _in >> c >> side_id;
-
-      // Store this pair of data in the vector
-      id_storage.push_back( std::make_pair(elem_id, side_id) );
+        // Store this pair of data in the vector
+        id_storage.push_back( std::make_pair(elem_id, side_id) );
+      }
+        else
+        {
+            // For the form of "Elset_1, S3"
+            _in.clear();
```

Comment:
Hmm... are you are using `clear()` here to clear some bad or fail bit on the stream that was set during the previous read? If you have tested this and it works, I guess it's OK, but my preference would be to call `std::getline` to extract the entire line from the stream and then do some string processing to figure out which type of sideset is present.

Reply:
Hi John - yes. We try to read in the `elem_id`, which is a number; if it failed, then we reset the fail bit and try to read as a string. Just to make sure I got it right: you are suggesting that I use `std::getline` to read the whole line. And then I will try to convert the string to `istream` & use `>>` to treat the first part as a number; if that failed, I will treat the first part as a string (`elset_name`). In this case, no fail bit resetting is required. Is that understanding correct?

Reply:
I edited the above paragraph a bit to make it clearer - you might want to check the webpage version to see the updates :)

Reply:
Hi @jwpeterson ! I just modified the code per my understanding. Please let me know what you think - or feel free to edit directly as well; I can always test them out before merging :)